libexec/cli/build.exec

```diff
@@ -255,6 +255,23 @@ case $SINGULARITY_BUILDDEF in
             zcat "$i" | (cd "$SINGULARITY_ROOTFS"; tar --exclude=dev/* -xf - ) || exit $?
         done
 
+        # If SINGULARITY_CREATE_HOME set and user home dir not in container
+        # then inject it if we are not root here.
+        if [ ! -z "${SINGULARITY_CREATE_HOME:-}" ]; then
+
+            if [ "$(id -u)" == "0" ]; then
+               message ERROR "Cannot use SINGULARITY_CREATE_HOME as root\n"
+               ABORT 255
+            fi
+
+            ROOTFS_HOME="${SINGULARITY_ROOTFS}${HOME}"
+            if [ ! -d "$ROOTFS_HOME" ]; then
+                message 1 "SINGULARITY_CREATE_HOME set. Creating missing $ROOTFS_HOME\n"
```

Comment:
Another idea is that it could be created on the fly as one of the layers (or in the metadata tar.gz) by the python, at least that way we have assurance it's extracted with rootfs and we don't have custom mkdir -p outside of that.

Reply:
That's a nice idea. I'll have to look into that. This was the quickest place I found to add it to satisfy a need to do a demo on Friday.

Reply:
If others think it's a good idea I can offer to do it - I spent a lot of time figuring out in memory tar creation so it wouldn't take super long. But it does bring up this larger (kind of cool?) question - can we give some greater options, at runtime, to modify what is being extracted? Without compromising safety? Is that even a good idea given that a container (is supposed to be) immutable?

Reply:
Nick Eggleston on the google groups list would probably support this - as he's hitting the same issue on 6.x. I'd guess it's quite widespread that RHEL 6.x is in use with a non standard home.

Reply:
just curious, why is there /home and /home2?

Reply:
On our systems /home is a local per-node location for some administrative accounts. Then there was /home1 where users had space, and it was replaced with /home2 when another storage system came along. On Linux environment at a previous university I recall we had people split by department like /home-dcs for CS folks /home-eng for the engineers.